lib/bademagic_module/bluetooth/connectstate.dart

```
@@ -0,0 +1,53 @@
+import 'package:badgemagic/bademagic_module/bluetooth/writestate.dart';
+import 'package:flutter_blue_plus/flutter_blue_plus.dart';
+import 'package:logger/logger.dart';
+import 'ble_state_interface.dart';
+import 'bletoast.dart';
+
+class ConnectState implements BleState {
+  final ScanResult scanResult;
+  final Logger logger = Logger();
+  final int maxRetries = 3;
+  BleStateToast toast = BleStateToast();
+
+  ConnectState({required this.scanResult});
+
+  @override
+  Future<BleState?> processState() async {
+    int attempt = 0;
+    bool connected = false;
+
+    while (attempt < maxRetries && !connected) {
+      try {
+        await scanResult.device.connect(autoConnect: false);
+        BluetoothConnectionState connectionState =
+            await scanResult.device.connectionState.first;
+
+        if (connectionState == BluetoothConnectionState.connected) {
+          logger.d("Device connected");
+          connected = true;
+          toast.successToast('Device connected successfully.');
+          return WriteState(device: scanResult.device);
+        } else {
+          logger.e("Failed to connect to the device");
```

Throw an error rather than logging